### DISCUSSION TITLE
Is the compiled object path stored within the nlmixr2est fit (and rxode2 object)?

### DISCUSSION THREAD

**Discussion by billdenney** [nlmixr2/nlmixr2est#429] [Q&A]
I'm running models in `nlmixr2targets`, and they are rerunning repeatedly without the model or data changing.  My best guess is that the MD5 hash is changing for some reason.  So, I'm trying to trace what may be changing within the model itself.  My first guess went to the path to the compiled object.  Might it change from run-to-run?

**Comment by mattfidler** [comment]
The path isn't stored in the object, so it shouldn't affect the md5 hash.  

In answer to your question; every time a R session starts it creates a new temporary directory. That temporary directory is where rxode2 creates a folder if `rxCreateCache()` has not been run.  So it will create a new directory for every R session.

However, if `rxCreateCache()` is created, everything is stored inside of the cache directory.  If the md5 hash of the parsed object matches it does not rerun the compile; it simply loads the dll.

Because CRAN policies on runtime-linking every rxode2 code is linked based created based on the time of the object created and the  md5 hash of the parsed model.  The best bet to get the right hash with a model function is likely:

```r
mod <- deparse1(as.function(ui))
```

This will register label changes as a change as well as parameter order changes, but it may fit your use-case.

Internally, rxode2 uses the normalized syntax model for caching and md5, which may not capture enough for you.
